File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        alphas = list("abcdefghijklmnopqrstuvwxyz")
        mors = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        a_to_m = {}
        for a,m in zip(alphas,mors):
            a_to_m[a] = m
        
        result = []
        for word in words :
            this = ""
            for w in word:
                this += a_to_m[w]
            result.append(this)
        
        return len(set(result))

File: 804-unique-morse-code-words/804-unique-morse-code-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function(words) {
    let m = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."][Symbol.iterator]();
    let a = "abcdefghijklmnopqrstuvwxyz"[Symbol.iterator]();
    let hashs = {};
    
    for (let i=0; i<26; i++) {
        hashs[a.next().value] = m.next().value
    }
    results = new Set(words.map((word) => word.split("").map((char) => hashs[char]).join("")))
    return results.size
};